CHANGELOG.md

```diff
@@ -5,6 +5,11 @@
 ### Behavior Changes
 
 * Disable log recycling by default.
+* `LogBatch::put` returns a `Result<()>` instead of `()`. It errs when the key is reserved for internal use.
```

Comment:
How about panicking inside to make API clean? As it's a static format and user can do nothing except panic.

Reply:
In this crate panic is only used for unrecoverable error. For your case user can choose to bubble the error to UI for example.

Reply:
I'm afraid not. It's used for raft log, it's an undefined error if continue processing while last logs failed to be written.

Reply:
It's a log batch method, not an engine method. Even if it's an engine method, returning an error means this operation can be retried by user without causing damage to existing data.